Makefile.defs

```diff
@@ -77,15 +77,17 @@ CILIUM_ENVOY_SHA=$(shell grep -o "FROM.*cilium/cilium-envoy:[0-9a-fA-F]*" $(ROOT
 GO_BUILD_LDFLAGS += -X "github.com/cilium/cilium/pkg/envoy.RequiredEnvoyVersionSHA=$(CILIUM_ENVOY_SHA)"
 
 # Use git only if in a Git repo, otherwise depend on file BPF_SRCFILES existing
+BPF_SRCFILES_IGNORE = bpf/.gitignore
 ifneq ($(wildcard $(dir $(lastword $(MAKEFILE_LIST)))/.git),)
-	BPF_FILES_EVAL = $(shell git ls-files $(ROOT_DIR)/bpf/ | grep -v .gitignore | tr "\n" ' ')
-	BPF_FILES ?= $(BPF_FILES_EVAL)
-	BPF_SRCFILES := $(subst ../,,$(BPF_FILES))
+	BPF_SRCFILES := $(shell git ls-files $(ROOT_DIR)/bpf/ | sort | tr "\n" ' ')
 else
-	BPF_SRCFILES = $(shell cat $(ROOT_DIR)/BPF_SRCFILES)
+	# this line has to be in-sync with bpf/.gitignore, please note usage of make patterns like `%.i`
+	BPF_SRCFILES_IGNORE += bpf/cilium-map-migrate bpf/cilium-probe-kernel-hz bpf/%.i bpf/%.s bpf/.rebuild_all
```

Comment:
Isn't it possible to read the content of the `bpf/.gitignore` file (and possibly the parent one) rather than duplicating the `.gitignore` file content here?

Reply:
The syntax is very different, we cannot just read it as is, even if we striped the comment - it has regex support (including negation), while `filter-out` macro supports make's own pattern syntax. We'd need to drag in some sort gitignore parser. I'd much rather just improve this by creating a go Tool that can be used by the loaded as well as during the build. In fact, I believe loaded already has most of logic.